Src/FluentAssertions/Equivalency/Selection/SelectMemberByPathSelectionRule.cs

```diff
@@ -7,32 +6,12 @@ namespace FluentAssertions.Equivalency.Selection;
 
 internal abstract class SelectMemberByPathSelectionRule : IMemberSelectionRule
 {
-    private string selectedPath;
-
-    protected SelectMemberByPathSelectionRule(string selectedPath)
-    {
-        this.selectedPath = selectedPath;
-    }
-
     public virtual bool IncludesMembers => false;
 
-    protected void SetSelectedPath(string path)
-    {
-        selectedPath = path;
-    }
-
     public IEnumerable<IMember> SelectMembers(INode currentNode, IEnumerable<IMember> selectedMembers,
         MemberSelectionContext context)
     {
-        string currentPath = currentNode.PathAndName;
-
-        // If we're part of a collection comparison, the selected path will not include an index,
-        // so we need to remove it from the current node as well.
-        if (!ContainsIndexingQualifiers(selectedPath))
```

Comment:
This is the part where I am not sure about the implications, since all tests are green